spec/spec_helper.rb

```diff
@@ -6,7 +6,13 @@
 module SpecHelper
   # @private
   def adapter
-    ENV["RAUTOMATION_ADAPTER"] && ENV["RAUTOMATION_ADAPTER"].to_sym || RAutomation::Adapter::Helper.default_adapter
+     adapter = ENV["RAUTOMATION_ADAPTER"] && ENV["RAUTOMATION_ADAPTER"].to_sym || RAutomation::Adapter::Helper.default_adapter
+
+     if adapter != :win32 && !Platform.is_x86?
```

Comment:
We can use `Adapter::Helper` method here now to detect if adapter is supported on the platform or not. No need to have separate logic in here.